parsing/parsing.go

```diff
@@ -115,30 +119,39 @@ func Parsing(task tasks.ParsingTask) (tasks.ParsingResult, error) {
 
 				select {
 				case res := <-app.Call("enqueue", "aggregate_host", t):
+					if res == nil {
+						logger.Errf("%s Task failed: %s", task.Id, common.ErrAppCall)
+						return
+					}
 					if res.Err() != nil {
-						logger.Errf("%s Task failed  %s", task.Id, res.Err())
+						logger.Errf("%s Task failed: %s", task.Id, res.Err())
 						return
 					}
 
 					var raw_res []byte
 					if err := res.Extract(&raw_res); err != nil {
-						logger.Errf("%s Unable to extract result. %s", task.Id, err.Error())
+						logger.Errf("%s Unable to extract result: %s", task.Id, err.Error())
 						return
 					}
 
-					key := fmt.Sprintf("%s;%s;%s;%s;%v",
-						task.Host, task.ParsingConfigName,
-						logName, k, task.CurrTime)
-					result[key] = raw_res
+					key := fmt.Sprintf("%s;%s", task.Host, k)
+					ch <- item{key: key, res: raw_res}
 					logger.Debugf("%s Write data with key %s", task.Id, key)
 				case <-time.After(deadline):
 					logger.Errf("%s Failed task %s", task.Id, deadline)
 				}
-			}(aggType, k, v, aggLogName, time.Second*time.Duration(task.CurrTime-task.PrevTime))
+			}(app, k, v, time.Second*time.Duration(task.CurrTime-task.PrevTime))
 		}
 	}
-	wg.Wait()
+	go func() {
```

Comment:
But without go func() all goroutines will be blocked on sending to `ch`
There will be no return from other goroutines and wg.Done cannot be called `for` become unreachable